### PR TITLE
Modal: Add expand and collapse functionality

### LINKED ISSUE #2521: Full Screen Modal
Based on the submission: https://trimble-arch.aha.io/ideas/ideas/MOD-I-71

To be added to Figma by Dafnee Z.

### PR DESCRIPTION
**Pull request by prashanthr6383**
## Description

References 

Fixes #2521 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-modal/modus-modal.scss [hunk 1]
```diff
@@ -54,6 +54,40 @@
           opacity: $modus-modal-close-hover-opacity;
         }
       }
+
+      .icon-expand {
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-modal/modus-modal.scss:58]: can you consider reusing the existing styles used for close button using CSS grouping?

Comment (prashanthr6383) [reply stencil-workspace/src/components/modus-modal/modus-modal.scss]: @prashanth-offcl grouping is done. I have used the predefined styles of close icon but still looks incorrect

File: stencil-workspace/src/components/modus-modal/modus-modal.scss [hunk 2]
```diff
@@ -56,6 +58,17 @@
...
     }
 
+    .header-resize-buttons {
+      align-items: center;
+      display: flex;
+      gap: $rem-8px;
+      justify-content: center;
+
+      .resize-button {
+        font-size: 0; // to center align and remove the additional vertical space
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-modal/modus-modal.scss:68]: I see the close icon still have the vertical space issue, can you change the class name and apply the same class to close icon
image

Comment (prashanthr6383) [reply stencil-workspace/src/components/modus-modal/modus-modal.scss:68]: okay

Comment (prashanthr6383) [reply stencil-workspace/src/components/modus-modal/modus-modal.scss:68]: @prashanth-offcl added please check

**Review by prashanth-offcl — APPROVED** [review]

**Pull request merged**